metron-analytics/metron-profiler-client/src/test/java/org/apache/metron/profiler/client/stellar/GetProfileTest.java

```diff
@@ -20,437 +20,182 @@
-                    .withClass(FixedLookback.class),
-            new Context.Builder()
-                    .with(Context.Capabilities.GLOBAL_CONFIG, () -> global)
-                    .build());
-  }
-
-  /**
-   * This method is similar to setup(), in that it sets up profiler configuration context,
-   * but only for the client.  Additionally, it uses periodDuration2, periodUnits2
-   * and saltDivisor2, instead of periodDuration, periodUnits and saltDivisor respectively.
-   *
-   * This is used in the unit tests that test the config_overrides feature of PROFILE_GET.
-   * In these tests, the context from @Before setup() is used to write the data, then the global
-   * context is changed to context2 (from this method).  Each test validates that a default read
-   * using global context2 then gets no valid results (as expected), and that a read using
-   * original context values in the PROFILE_GET config_overrides argument gets all expected results.
-   *
-   * @return context2 - The profiler client configuration context created by this method.
-   *    The context2 values are also set in the configuration of the StellarStatefulExecutor
-   *    stored in the global variable 'executor'.  However, there is no API for querying the
-   *    context values from a StellarStatefulExecutor, so we output the context2 Context object itself,
-   *    for validation purposes (so that its values can be validated as being significantly
-   *    different from the setup() settings).
-   */
-  private Context setup2() {
-    state = new HashMap<>();
+    // the VERBOSE_PROFILE function that will be tested
+    function = new GetProfile();
+    function.withProfilerClientFactory(globals -> profilerClient);
 
     // global properties
-    Map<String, Object> global = new HashMap<String, Object>() {{
-      put(PROFILER_HBASE_TABLE.getKey(), tableName);
-      put(PROFILER_COLUMN_FAMILY.getKey(), columnFamily);
-      put(PROFILER_HBASE_TABLE_PROVIDER.getKey(), MockHBaseTableProvider.class.getName());
-      put(PROFILER_PERIOD.getKey(), Long.toString(periodDuration2));
-      put(PROFILER_PERIOD_UNITS.getKey(), periodUnits2.toString());
-      put(PROFILER_SALT_DIVISOR.getKey(), Integer.toString(saltDivisor2));
-    }};
-
-    // create the modified context
-    Context context2 = new Context.Builder()
-            .with(Context.Capabilities.GLOBAL_CONFIG, () -> global)
+    globals = new HashMap<>();
+    Context context = new Context.Builder()
+            .with(Context.Capabilities.GLOBAL_CONFIG, () -> globals)
             .build();
 
     // create the stellar execution environment
-    executor = new DefaultStellarStatefulExecutor(
-            new SimpleFunctionResolver()
-                    .withClass(GetProfile.class)
-                    .withClass(FixedLookback.class),
-            context2);
+    functionResolver = new SimpleFunctionResolver()
+            .withClass(FixedLookback.class)
+            .withInstance(function);
```

Comment:
Being able to inject the function instance that we want to test makes writing these tests much simpler.  This allows us to plug-in a mock `ProfilerClient` used by the function under test.